src/sequence.hpp

```diff
@@ -54,6 +52,7 @@ class Sequence {
     std::mutex& getMutex();
     bool isTerminated() const;
     void setTerminated();
+    void updateLastActivityTime();
```

Comment:
I think we should keep this method private or protected and get back to calling it from the inside of some public method.